source/mir/random/ndvariable.d

```diff
@@ -279,6 +280,114 @@ nothrow @safe version(mir_random_test) unittest
     assert(fabs(x[0] + x[1] + x[2] - 1) < 1e-10);
 }
 
+/++
+Multinomial distribution.
++/
+struct MultinomialVariable(U:size_t, T)
+    if (isFloatingPoint!T)
+{
+    import mir.random.variable : binomialVar;
+
+    ///
+    enum isNdRandomVariable = true;
+    ///
+    alias Element = uint;
+
+    ///
+    const(T)[] probs;
+    size_t N;
+
+    /++
+    Params:
+        probs = probabilities of the multinomial distribution
+        N = Number of rolls
+    Constraints: `sum(probs[i]) <= 1`
+    +/
+    this()(size_t N, const(T)[] probs)
+    {
+        this.N = N;
+        this.probs = probs;
+    }
+
+    ///
+    pragma(inline, false)
+    void opCall(G)(scope ref G gen, scope U[] result)
+        if (isSaturatedRandomEngine!G)
+    {
+        double norm = 0.0;
+        double sum_p = 0.0;
+
+        uint sum_n = 0;
+
+
+        /// Makes sure probabilities add up to one, by calculating a normalization term
+        foreach(k, p; this.probs)
+        {
+            norm += p;
+        }
+
+        foreach(k, p; this.probs)
+        {
+            if (p > 0.0)
+            {
+                auto rv = binomialVar!T(this.N - sum_n, p / (norm - sum_p));
+                result[k] = cast(uint)rv(gen);
+
+            }
+            else
+            {
+                result[k] = 0;
+            }
+
+            sum_p += p;
+            sum_n += result[k];
+        }
+
+
+    }
+    /// ditto
+    void opCall(G)(scope G* gen, scope U[] result)
+        if (isSaturatedRandomEngine!G)
+    {
+        pragma(inline, true);
+        return opCall(*gen, result);
+    }
+}
+
+/// ditto
+MultinomialVariable!(U,T) multinomialVar(U, T)(in U N, in T[] probs)
```

Comment:
`MultinomialVariable!(T) multinomialVar(T)(in size_t N, in T[] probs)`

Reply:
Done